libsplinter/src/admin/service/mod.rs

```diff
@@ -524,6 +524,24 @@ impl Service for AdminService {
     }
 }
 
+fn handle_peer_manager_notification(
+    notification: PeerManagerNotification,
+    mut admin_shared: MutexGuard<AdminServiceShared>,
+) {
+    match notification {
+        PeerManagerNotification::Connected { peer } => {
+            if let Err(err) = admin_shared.on_peer_connected(&peer) {
+                debug!("Peer {} has connected", peer);
```

Comment:
Should this be outside the if statement?

Reply:
fixed